File: 2026-08/day0054/f6_rebuild.js
let task_id = null;

$(".edit-btn").on("click", function () {
    task_id = $(this).data("task-id");

    $.ajax({
        url: `/tasks/${task_id}/data`,
        type: "GET",
        success: function (response) {
            $("#edit-title").val(response["title"]);

            if (response["done"] === true) {
                $("#edit-status").text("완료");
            } else {
                $("#edit-status").text("미완료");
            }
        }
    });
});

$("#save-btn").on("click", function () {
    $.ajax({
        url: `/tasks/${task_id}/title`,
        type: "POST",
        data: {
            title: $("#edit-title").val(),
        },
        success: function (response) {
            $(`#task-title-${task_id}`).text(response.title)
        }
    });
});

File: 2026-08/day0054/f6_rebuild.py
from flask import Flask, request, jsonify

app = Flask(__name__)

tasks = {
    "2": {
        "title": "Flask 복습",
        "done": False
    },
    "6": {
        "title": "Ajax 정리",
        "done": True
    }
}



@app.route("/tasks/<task_id>/data")
def getTask(task_id):
    return jsonify(tasks[task_id])


@app.route("/tasks/<task_id>/title", methods=["POST"])
def postTask(task_id):
    new_title = request.form.get("title")

    if task_id in tasks:
        tasks[task_id]["title"] = new_title

        return jsonify(tasks[task_id])


